src/agent_workflow_server/agents/adapters/llamaindex.py

```diff
@@ -59,7 +70,15 @@ async def astream(self, run: Run):
             handler.ctx.send_event(HumanResponseEvent(response=user_data))
 
         async for event in handler.stream_events():
-            self.contexts[run["thread_id"]] = handler.ctx.to_dict()
+            if checkpoints is None:
+                checkpoints = []
+
+            checkpoints.append(
```

Comment:
By doing this, you are adding checkpoints also for the intermediate events, is it correct?  Is this for consistency with Langgraph? Otherwise you could just add the checkpoint at the end of the run (i.e. what you do at line 93.

Reply:
For consistency. But if we don't insist on it, i can remove it :)